src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1441,6 +1441,74 @@ def accept_market_ordering_terms(cmd, urn=None, publisher=None, offer=None, plan
 # endregion
 
 
+def _terms_prepare(cmd, urn, publisher, offer, plan):
+    usage_err = 'usage error: --plan STRING --offer STRING --publish STRING |--urn STRING'
+    if urn:
+        if any([publisher, offer, plan]):
+            raise CLIError(usage_err)
+        publisher, offer, _, _ = urn.split(':')
```

Comment:
add try/except for split(':')

Reply:
We can also add a validator to do argument checks.